Adapter/PhpcrOdmAdapter.php

```diff
@@ -190,12 +203,42 @@ public function getReferringAutoRoutes($contentDocument)
     public function findRouteForUri($uri, UriContext $uriContext)
     {
         $path = $this->getPathFromUri($uri);
-
-        return $this->dm->find(null, $path);
+        $node = $this->dm->find(null, $path);
+        if ($node instanceof AutoRouteInterface) {
+            return $node;
+        }
+        return false;
     }
 
     private function getPathFromUri($uri)
     {
         return $this->baseRoutePath . $uri;
     }
+
+    /**
+     * @param Generic $node
+     * @param object $contentDocument
+     * @param string $autoRouteTag
+     * @param string $routeType
+     * @return AutoRouteInterface
+     */
+    private function convertGenericNodeInAutoRouteNode(Generic $node, $contentDocument, $autoRouteTag, $routeType)
```

Comment:
rename to `migrateGenericToAutoRoute` ? Add a comment:

````
Convert the given generic document to an auto route document, migrating any children which it may have.
````